client/src/AjaxButtons.jsx

```diff
@@ -0,0 +1,191 @@
+          console.log('toggle off',format)
+        this.setState({
+          funnyCount: format.funnyCount,
+          funnyColor: format.funnyColor,
+          funnyToggle: format.funnyToggle,
+        })
+      })
+      .catch((error)=>{
+        console.log(error)
+      })
+    }
+  }
+
+  coolUpdate(){
+    this.setState({
+      voteButtonResponse: "Thanks for your vote!",
+      responseColor: "green"
+    })
+    if (this.state.coolToggle === 0){
+      axios.put(`/api/restaurant/${this.props.review.id}`, {
+        name: this.props.review.name,
+        coolCount: this.state.coolCount + 1,
+        coolToggle: 1,
+        coolColor: 'blue'
+      })
+      .then((response)=>{
+        var format = JSON.parse(response.config.data)
+          console.log('toggle on',format)
+        this.setState({
+          coolCount: format.coolCount,
+          coolColor: format.coolColor,
+          coolToggle: format.coolToggle,
+        })
+      })
+      .catch((error)=>{
+        console.log(error)
+      })
+    }
+    else {
+      axios.put(`/api/restaurant/${this.props.review.id}`, {
+        name: this.props.review.name,
+        coolCount: this.state.coolCount -1,
+        coolToggle: 0,
+        coolColor: 'white'
+      })
+      .then((response)=>{
+        var format = JSON.parse(response.config.data)
+          console.log('toggle off',format)
+        this.setState({
+          coolCount: format.coolCount,
+          coolColor: format.coolColor,
+          coolToggle: format.coolToggle,
+        })
+      })
+      .catch((error)=>{
+        console.log(error)
+      })
+    }
+  }
+  render(){
+    return(
+    <div className="buttonbody">
+       <div className="buttonresponse" style={{'color':this.state.responseColor}}>{this.state.voteButtonResponse}</div>
+          <div>
+            <div className="buttonuseful" >
+              <button style={{'backgroundColor':this.state.usefulColor}} onClick={this.usefulUpdate}><img className="buttonpress" src="https://dumielauxepices.net/sites/default/files/styles/225x120/public/light-bulb-clipart-logo-664919-5076127.png?itok=ZoxF3t1z" />  Useful {this.state.usefulCount}</button>
+            </div>
+            <div className="buttonfunny">
+              <button style={{'backgroundColor':this.state.funnyColor}} onClick={this.funnyUpdate}><img className="buttonpress" src="https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/120/apple/155/face-with-tears-of-joy_1f602.png"/>  Funny {this.state.funnyCount}</button>
+            </div>
+            <div className="buttoncool">
+               <button style={{'backgroundColor':this.state.coolColor}} onClick={this.coolUpdate}><img className="buttonpress" src="https://www.simplystamps.com/media/catalog/product/cache/5/image/600x600/9df78eab33525d08d6e5fb8d27136e95/E/M/EMOJI_05.png"/>  Cool {this.state.coolCount}</button>
+            </div>
+          </div>
+    </div>
+    )
+  }
+}
+
+export default AjaxButtons;
```

Comment:
I feel that your update functions repeat a lot of code. Is there a way to compartmentalize some of this?

What does axios do?  I've never heard of it before.

Reply:
Also, if you're wanting to use these specific images,  I would reccomend using an S3 bucket to store them, and reference them from there.  It would prevent using these really long link urls and make the image safe in case the source ever changes it.